compiler/fir/java/src/org/jetbrains/kotlin/fir/java/enhancement/EnhancementSignatureParts.kt

```diff
@@ -219,6 +224,14 @@ internal class EnhancementSignatureParts(
                 )
             }
 
+        if (attributesCache.getValue(this) == ConeAttributes.Empty) {
+            val isJavaPlatformClass =
+                this is FirJavaTypeRef && this.toFqName() != null && JavaToKotlinClassMap.isJavaPlatformClass(this.toFqName()!!)
+            if (isJavaPlatformClass) {
```

Comment:
According to IR text test data, there are more places that need to record `@FlexibleNullability`, e.g., type parameter, or many other declaration places. I'm just not that familiar with `:fir:java` module yet. :\

Reply:
This check are suspicious for me

`if (attributesCache.getValue(this) == ConeAttributes.Empty)`

I think you are trying to check something like "There is no enhanced nullability", aren't you? May be it's better to modify the check accordingly? (just in case some other attributes will appear in future, which can break logic here)

Reply:
About more places, well, it's a good place to start.

Reply:
Also, why this check `if (isJavaPlatformClass)` is needed? As I know, all Java types with flexible nullability should have this annotation...

Reply:
Right, my intention was to check something like that, and right, again, that this is quite fragile.

Re: platform class, I saw many previous commits mentioned platform type for `@FlexibleNullability`, so I assumed it's bound to platform type only. :p

Anyway, this is completely rewritten, and not needed anymore.